test/Microsoft.AspNet.Mvc.FunctionalTests/TryValidateModelTest.cs

```diff
@@ -87,5 +81,37 @@ public class TryValidateModelTest
             var body = await response.Content.ReadAsStringAsync();
             Assert.Equal("{}", body);
         }
+
+        [Fact]
+        public async Task TryValidateModel_CollectionsModel_ReturnsErrorsForInvalidProperties()
```

Comment:
The ```TryValidateModelAfterClearingValidationErrorInParameter``` test explicitly clears the ModelStateDictionary while the one you are adding should implicitly do that. You s hould either modify the other test or add a separate one.